int32 num_lines
Single_Line[] line
